hnn_core/gui/gui.py

```diff
@@ -888,6 +888,43 @@ def _prepare_upload_file(path):
     return uploaded_value
 
 
+def _update_nested_dict(original, new, skip_none=True):
+    """ Updates dictionary values from another dictionary
+
+    Will update nested dictionaries in the structure. New items from the
+    update dictionary are added and omitted items are retained from the
+    original dictionary. By default, will not pass None values from the update
+    dictionary.
+
+    Parameters
+    ----------
+    original : dict
+        Dictionary to update
+    new : dict
+        Dictionary with new values for updating
+    skip_none : bool, default True
+        None values in the new dictionary are not passed to the updated
+        dictionary by when True. If False None values will be passed to the
+        updated dictionary.
+
+    Returns
+    -------
+
+    """
+    updated = original.copy()
+    for key, value in new.items():
```

Comment:
Would this behave correctly if an empty dictionary is passed in new? Like say you wanted to clear the data in one of the dictionaries

Reply:
An empty dict would not change the original because you must have a matching key to update. But if you had matching keys and had null values you could update with null values by setting `skip_none` to False. 

Thanks for brining up this question though. This reminds me that I need to make a change to the logic. Right now it will not update 0 values in the 'new' dict by default because they evaluate as False at 920. It should be changed to an explicit `value != None`